app/data/Owners.scala

```diff
@@ -0,0 +1,37 @@
+package data
+
+import model._
+
+trait Owners {
+  def stacks: Set[(String, SSA)]
+
+  def all: Set[Owner] = stacks
+    .groupBy(_._1)
+    .map { case (ownerId, stacks) =>
+      Owner(ownerId, stacks.map(_._2))
+    }.toSet
+
+  def find(id: String): Option[Owner] = all.find(_.id == id)
+
+  def find(ssa: SSA): Option[Owner] = all.find(_.hasSSA(ssa))
+
+  def forStack(stackName: Option[String], stageName: Option[String], appName: Option[String]): Option[Owner] = all
+    .find(_.hasSSA(SSA(stackName, stageName, appName)))
+    .orElse(all.find(_.hasSSA(SSA(stackName, stageName))))
+    .orElse(all.find(_.hasSSA(SSA(stackName))))
+    .orElse(all.find(_.hasSSA(SSA())))
+}
+
+object Owners extends Owners {
+
+  override def stacks: Set[(String, SSA)] = Set(
+    "phil.wills" -> SSA(), // Default owner of all stacks
```

Comment:
I think that this is the right way around to do it. As @TBonnin says it is possible to reverse this for doing work on it. I notice that there is a unit test to assert that there is only one owner per stack 👍